grc/core/params/dtypes.py

```diff
@@ -40,6 +40,11 @@ class ValidateError(Exception):
 def validate_block_id(param,black_listed_ids):
     value = param.value
     # Can python use this as a variable?
+
+    is_exempt_from_id_validation = hasattr(param.parent_block, 'exempt_from_id_validation')
+    # Grant exemption to epy blocks and modules
+    if is_exempt_from_id_validation:
+        return value
```

Comment:
Hm, not sure about this one. The other checks are still useful.
I think it would be better to use this flag only for disabling the blacklist check

Reply:
What do you mean? Would you like me to have the `getattr` check _and_ also check the block key like in v3?

Reply:
Am I correct that you mean something like the code below by "the other checks"?
```python
is_exempt_from_id_validation = getattr(param.parent_block, 'exempt_from_id_validation', False)
if is_exempt_from_id_validation and re.match(r'^epy_(block|module)$', param.parent_block.key):
    return value
```
I cant appreciate how having that additional block key check would be beneficial. Wouldn't having this additional check  complicate maintenance? Because the bug will come back should the block key or name change for whatever reason. I feel a single `getattr` check would be better. Makes it easier to add or remove blocks to the exemption too should the need ever arise. What do you think?

Reply:
Ok I gave it even more thought. I think I get what you mean now by "disabling the blacklist check"! Let me make a v7. Im so sorry Im commenting so much! I just cant help but try to think of better ways to make this patch

Reply:
Yes, that is what I meant. I should have made that clearer in my comment... Sorry about that